docs/deploy/Deploy-FhirConverterService.bicep

```diff
@@ -0,0 +1,193 @@
+
+@description('If set to true, security will be enabled on the API endpoint.')
+param securityEnabled bool = false
+
+@description('List of audiences that the security token is intended for.')
+param securityAuthenticationAudiences array = []
+
+@description('Issuing authority of the JWT token.')
+param securityAuthenticationAuthority string = ''
+
+@description('Tag of the image to deploy.')
+param imageTag string
+
+@description('Timestamp to append to container name. Defaults to time of deployment.')
+param timestamp string = utcNow('yyyyMMddHHmmss')
+
+// Security configuration
+var securityEnabledConfigName = 'ConvertService__Security__Enabled'
+var securityAuthenticationAudiencesConfigNamePrefix = 'ConvertService__Security__Authentication__Audiences__'
+var securityAuthenticationAuthorityConfigName = 'ConvertService__Security__Authentication__Authority'
+var securityConfiguration = [
+  {
+    name: securityEnabledConfigName
+    value: string(securityEnabled)
+  }
+  {
+    name: securityAuthenticationAuthorityConfigName
+    value: securityAuthenticationAuthority
+  }
+]
+
+var securityAuthenticationAudiencesConfig = [for (audience, i) in securityAuthenticationAudiences: {
+    name: '${securityAuthenticationAudiencesConfigNamePrefix}${i}'
+    value: audience
+}]
+
+// Template hosting configuration
+var storageEnvironmentSuffix = az.environment().suffixes.storage
+var blobTemplateHostingConfigurationName = 'TemplateHosting__StorageAccountConfiguration__ContainerUrl'
+var blobTemplateHostingConfigurationValue = 'https://${templateStorageAccountName}.blob.${storageEnvironmentSuffix}/${templateStorageAccountContainerName}'
+var blobTemplateHostingConfiguration = [
+  {
+    name: blobTemplateHostingConfigurationName
+    value: blobTemplateHostingConfigurationValue
+  }
+]
+
+// Telemetry configuration
+var applicationInsightsName = '${envName}-ai'
+resource applicationInsights 'Microsoft.Insights/components@2020-02-02' existing = {
+  name: applicationInsightsName
+}
+
+resource applicationInsightsUAMI 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing = {
+  name: '${applicationInsightsName}-mi'
+}
+
+var appInsightsConnectionStringConfigurationName = 'ConvertService__Telemetry__AzureMonitor__ApplicationInsightsConnectionString'
+var appInsightsConnectionString = applicationInsights.properties.ConnectionString
+var appInsightsUAMIClientIdConfigurationName = 'ConvertService__Telemetry__AzureMonitor__ManagedIdentityClientId'
+var appInsightsUAMIClientId = applicationInsightsUAMI.properties.clientId
+var telemetryConfiguration = [
+    {
+        name: appInsightsConnectionStringConfigurationName
+        value: appInsightsConnectionString
+    }
+    {
+        name: appInsightsUAMIClientIdConfigurationName
+        value: appInsightsUAMIClientId
+    }
+]
+
+// Get Container Apps Environment
+resource containerAppEnvironment 'Microsoft.App/managedEnvironments@2023-05-01' existing = {

```

Comment:
nit: instead of fetching resources here can we take the necessary properties as required parameters to this template instead? While this assumption that this resource is created is right in this case, I think not having this dependency in this template but rather in the parent template invoking this makes it more clear?